### PR TITLE
[JS_13] Made some changes according to Oleg's comments

### PR DESCRIPTION
**Pull request by oonufrienko**
1) Changed the first "if" statement
2) Made changes according to the new data structure
3) Move part of the code to method implementation

### CODE REVIEW

File: js_13/index.html [hunk 1]
```diff
@@ -21,25 +21,23 @@ <h3 class="title">Тест по программированию</h3>
...
-                        </label>
-                      </div>
-                    </li>
+              <%=questions[i].questionTitle%>
+              <ul class="answers">
+              <% questions[i].answers.forEach( function(elem) { %>
+                <li>
+                  <div class="custom-control custom-checkbox">
+                    <input class="custom-control-input" type="checkbox" id=<%="customCheck" + elem.id%> >
+                    <label class="custom-control-label" for=<%="customCheck" + elem.id%> >
```
Comment (olegzmiuk) [inline js_13/index.html]: it should be
```
for="<%='customCheck' + elem.id%>">
```

Comment (olegzmiuk) [reply js_13/index.html]: same for `custom-control-input`

File: js_13/js/script.js [hunk 2]
```diff
@@ -2,76 +2,71 @@
 
 $(function(){
 
-  if (localStorage.getItem('sectionQA')) {
-
-    var sectionQA = localStorage.getItem('sectionQA');
+  if (!localStorage.getItem('questions')) { return; }
 
+    var questions = JSON.parse(localStorage.getItem('questions'));
```
Comment (olegzmiuk) [inline js_13/js/script.js]: don't forget to format code properly

File: js_13/js/script.js [hunk 3]
```diff
@@ -2,76 +2,71 @@
...
-      $('.result-modal .btn-close').on('click', function () {
-        hideModal();
-      });
-
-    });
-
+      return results;
+    }
 
     function showModal(result_obj) {
```
Comment (olegzmiuk) [inline js_13/js/script.js]: you should follow one naming style
in your case it should be camelCase `resultObj`

File: js_13/js/script.js [hunk 4]
```diff
@@ -2,76 +2,71 @@
...
-          correctAnswer = index - (sectionNumber * n);
-          if (correctAnswers.answerId[sectionNumber] === correctAnswer) {
+          sectionNumber = parseInt(liIndex / questions.length);
+          answerIndex = liIndex - (sectionNumber * questions.length);
+  
+          if (questions[sectionNumber].answers[answerIndex].isCorrect) {
             results.id[sectionNumber] = 'верно';
           } else {
-            results.id[sectionNumber] = 'не верно';
+              results.id[sectionNumber] = 'не верно';
```
Comment (olegzmiuk) [inline js_13/js/script.js]: why this two spaces is needed here? they are breaking the formatting

File: js_13/js/script.js [hunk 5]
```diff
@@ -2,76 +2,71 @@
...
+    $('.btn').on('click', function(){      
+      showModal(defineResults('.answers li input'));
+
+      $('.result-modal .btn-close').on('click', function () {
+        hideModal();
+      });
+      
+    });
+    
+    function defineResults(listElement) {
```
Comment (olegzmiuk) [inline js_13/js/script.js]: if you passing the string, it's better to name parameter like `listElementSelector` to make if more obvious

**Comment by oonufrienko** [comment]
@salieri328 please take a look at the changes I've made.

File: js_13/js/script.js [hunk 6]
```diff
@@ -2,76 +2,68 @@
...
+    $( listElementSelector ).each(function( liIndex ) {
+      if ( $(this).prop('checked') ) {
+        sectionNumber = parseInt(liIndex / questions.length);
+        answerIndex = liIndex - (sectionNumber * questions.length);
+  
+        if ( questions[sectionNumber].answers[answerIndex].isCorrect ) {
+          resultObj.answer[sectionNumber] = 'верно';
+        } else {
+          resultObj.answer[sectionNumber] = 'не верно';
+        }
```
Comment (olegzmiuk) [inline js_13/js/script.js]: also please note that such simple `if/else` statements can be rewritten with use of ternary operator
```javascript
resultObj.answer[sectionNumber] = questions[sectionNumber].answers[answerIndex].isCorrect ? 'верно' : 'не верно';
```
or
```javascript
resultObj.answer[sectionNumber] = questions[sectionNumber].answers[answerIndex].isCorrect
  ? 'верно'
  : 'не верно';
```

**Review by olegzmiuk — APPROVED** [review]

File: js_13/js/script.js (hunk 6 above)
Comment (oonufrienko) [reply js_13/js/script.js]: Got it. Thanks a lot 👍

**Pull request merged**